qcengine/programs/qchem.py

```diff
@@ -0,0 +1,254 @@
+        """
+
+        # Collect all input files and update with extra_infiles
+        infiles = inputs["infiles"]
+        if extra_infiles is not None:
+            infiles.update(extra_infiles)
+
+        binary_files = [os.path.join("savepath", x) for x in ["99.0", "131.0", "132.0"]]
+
+        # Collect all output files and extend with with extra_outfiles
+        outfiles = ["dispatch.out"]
+        if extra_outfiles is not None:
+            outfiles.extend(extra_outfiles)
+
+        # Replace commands with extra_commands if present
+        commands = inputs["commands"] + ["savepath"]
+        if extra_commands is not None:
+            commands = extra_commands
+
+        envs = self._get_qc_path()
+
+        with temporary_directory(parent=inputs["scratch_directory"], suffix="_qchem_scratch") as tmpdir:
+            envs["QCSCRATCH"] = tmpdir
+            bdict = {x: None for x in binary_files}
+
+            with disk_files({}, bdict, cwd=tmpdir, as_binary=binary_files):
+                exe_success, proc = execute(commands,
+                                            infiles=infiles,
+                                            outfiles=outfiles,
+                                            scratch_name=scratch_name,
+                                            scratch_directory=tmpdir,
+                                            scratch_messy=scratch_messy,
+                                            timeout=timeout,
+                                            environment=envs)
+
+            proc["outfiles"].update({os.path.split(k)[-1]: v for k, v in bdict.items()})
+
+        if "Thank you very much for using Q-Chem" not in proc["outfiles"]["dispatch.out"]:
+            exe_success = False
+
+        # QChem does not create an output file and only prints to stdout
+        return exe_success, proc
+
+    def build_input(self, input_model: 'ResultInput', config: 'JobConfig',
+                    template: Optional[str] = None) -> Dict[str, Any]:
+
+        # Check some bounds on what cannot be parsed
+        if "ccsd" in input_model.method.lower() or "ccd" in input_model.method.lower():
+            raise InputError("Cannot handle CC* methods currently.")
+
+        # Build keywords
+        keywords = {k.upper(): v for k, v in input_model.keywords.items()}
+        keywords["INPUT_BOHR"] = "TRUE"
+        keywords["MEM_TOTAL"] = str(int(config.memory * 1024))  # In MB
+
+        if input_model.driver == "energy":
+            keywords["JOBTYPE"] = "sp"
+        elif input_model.driver == "gradient":
+            keywords["JOBTYPE"] = "force"
+        elif input_model.driver == "hessian":
+            keywords["JOBTYPE"] = "freq"
+        else:
+            raise InputError(f"Driver of type {input_model.driver} is not yet supported.")
+
+        if input_model.molecule.fix_com or input_model.molecule.fix_orientation:
+            keywords["NO_REORIENT"] = "TRUE"
```

Comment:
if you come across that `no_reorient` discussion again, please point me at it -- I wasn't able to find anything specific on this vs. `sym_ignore`

Reply:
I couldn't find it after extensive googling.